src/node_buffer.cc

```diff
@@ -639,11 +639,11 @@ void Fill(const FunctionCallbackInfo<Value>& args) {
                                     enc,
                                     nullptr);
     // This check is also needed in case Write() returns that no bytes could
-    // be written.
-    // TODO(trevnorris): Should this throw? Because of the string length was
-    // greater than 0 but couldn't be written then the string was invalid.
+    // be written. If no bytes could be written, then throw because the string
+    // is invalid. Silently failing could lead to buffers with uninitialized
+    // memory.
     if (str_length == 0)
-      return;
+      return env->ThrowError("Buffer could not be filled");
```

Comment:
I'm not super familiar with this code but could this return a non-undefined value and then throw in JS? Then we could have an error code for it and document it?

Reply:
Throwing from JavaScript would be ideal